kiva/fonttools/_scan_parse.py

```diff
@@ -224,13 +224,16 @@ def _ttf_font_property(fpath, font, face_index=0):
     #  Weights are: 100, 200, 300, 400 (normal: default), 500 (medium),
     #    600 (semibold, demibold), 700 (bold), 800 (heavy), 900 (black)
     #    lighter and bolder are also allowed.
-    weight = None
-    for w in weight_dict.keys():
-        if style_prop.find(w) >= 0:
-            weight = w
-            break
+    # We prefer weight values from the OS/2 table, if available, otherwise
+    # infer from the "name" table's style
+    weight = props.get("weight")
     if not weight:
```

Comment:
I'm pretty sure this was meant to be `if weight is None:`, just with the bad habits of years past.